src/test/java/api/requests/RequestScheduledNoticesTests.java

```diff
@@ -0,0 +1,342 @@
+package api.requests;
+
+import static api.support.builders.RequestBuilder.OPEN_NOT_YET_FILLED;
+import static api.support.matchers.TextDateTimeMatcher.isEquivalentTo;
+import static org.folio.circulation.domain.representations.RequestProperties.HOLD_SHELF_EXPIRATION_DATE;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.junit.MatcherAssert.assertThat;
+
+import java.net.MalformedURLException;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.awaitility.Awaitility;
+import org.folio.circulation.domain.policy.Period;
+import org.folio.circulation.support.http.client.IndividualResource;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.LocalDate;
+import org.junit.Before;
+import org.junit.Test;
+
+import api.support.APITests;
+import api.support.builders.CheckInByBarcodeRequestBuilder;
+import api.support.builders.HoldingBuilder;
+import api.support.builders.ItemBuilder;
+import api.support.builders.NoticeConfigurationBuilder;
+import api.support.builders.NoticePolicyBuilder;
+import api.support.builders.RequestBuilder;
+import api.support.fixtures.ItemExamples;
+import api.support.http.InventoryItemResource;
+import io.vertx.core.json.JsonObject;
+
+public class RequestScheduledNoticesTests extends APITests {
+
+  private UUID templateId = UUID.randomUUID();
+  private InventoryItemResource item;
+  private IndividualResource requester;
+  private IndividualResource pickupServicePoint;
+
+  @Before
+  public void beforeEach()
+    throws MalformedURLException,
+    InterruptedException,
+    TimeoutException,
+    ExecutionException {
+
+    ItemBuilder itemBuilder = ItemExamples.basedUponSmallAngryPlanet(
+      materialTypesFixture.book().getId(), loanTypesFixture.canCirculate().getId());
+    HoldingBuilder holdingBuilder = itemsFixture.applyCallNumberHoldings(
+      "CN",
+      "Prefix",
+      "Suffix",
+      Collections.singletonList("CopyNumbers"));
+
+    item = itemsFixture.basedUponSmallAngryPlanet(itemBuilder, holdingBuilder);
+    requester = usersFixture.steve();
+    pickupServicePoint = servicePointsFixture.cd1();
+
+  }
+
+  @Test
+  public void requestExpirationUponAtNoticeShouldBeScheduledOnRequestCreate()
```

Comment:
What happens when a request is not set to expire? Is there a test for that and this event?

Reply:
In this case scheduled notice should not be scheduled. Added a test case.